cmd/bootnode/p2p.go

```diff
@@ -0,0 +1,78 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package bootnode
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	relaylog "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/host"
+	rcmgr "github.com/libp2p/go-libp2p/p2p/host/resource-manager"
+	"github.com/libp2p/go-libp2p/p2p/protocol/circuitv2/relay"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/p2p"
+)
+
+// startP2P returns a started libp2p host or an error.
+func startP2P(ctx context.Context, config Config, key *ecdsa.PrivateKey) (host.Host, error) {
+	if config.RelayLogLevel != "" {
+		if err := relaylog.SetLogLevel("relay", config.RelayLogLevel); err != nil {
+			return nil, errors.Wrap(err, "set relay log level")
+		}
+	}
+
+	// Increase resource limits
+	limiter := rcmgr.DefaultLimits
+	limiter.SystemBaseLimit.ConnsInbound = config.MaxConns
+	limiter.SystemBaseLimit.FD = config.MaxConns
+	limiter.TransientBaseLimit = limiter.SystemBaseLimit
```

Comment:
nit: add a comment for each of these limits

Reply:
I just moved this. You are are welcome to add the comments if you want.